pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -175,12 +175,16 @@ func determineSeed(shoot *gardencorev1beta1.Shoot, seedLister gardencorelisters.
 		return nil, err
 	}
 
-	filteredSeeds, err := filterSeedsMatchingSeedSelector(cloudProfile, seedList)
+	seedsMatchingCloudProfileSelector, err := filterSeedsMatchingSeedSelector(seedList, cloudProfile.Spec.SeedSelector, "CloudProfile")
+	if err != nil {
+		return nil, err
+	}
+	seedsMatchingShootSelector, err := filterSeedsMatchingSeedSelector(seedsMatchingCloudProfileSelector, shoot.Spec.SeedSelector, "Shoot")
```

Comment:
Hmm, this will yield an unhelpful message, if there are Seeds that are matching the `seedSelector` of the Shoot, but not the one of the CloudProfile.
I.e. it will say "none out of the <n> seeds has the matching labels required by seed selector of 'Shoot'" although, all Seeds might actually match the Shoot's selector.

Can you evaluate both selectors individually and then combine the filtered set of Seeds here to be able to give a more helpful error message?

Reply:
I haven't really understood what you are suggesting, can you explain again? The selectors are already evaluated individually. Also, I'm not too sure whether that's really worth the "effort" because there are other criteria where the seed selector labels may match but other preconditions don't. For example, I can have both an AWS and GCP seed with the same `foo=bar` labels, but when creating a GCP shoot with `foo=bar` seed selector then the AWS seed is not even considered. Isn't the `CloudProfile` filtering of the same quality?